R/paramatch.r

```diff
@@ -1,49 +1,73 @@
 #'
 #' This conducts full word matching only.  Substring matches do not count towards the total.
 #'
-#' @param input an atomic vector. The paragraph of text to analyse.
+#' @param dataframe
+#' @param col character. The column name pertaining to paragraphs for matching.
 #' @param top_n_words integer.  The number of top strings to count in each line.
 #' @param suppress.input logical.  Suppresses the Input column of the output data frame.
+#' @param blacklist character. Prevents undesired words participating int he output.
 #' @return numeric. Number of occurances of the particular word in each line.
-#' @export
+#' @author Mike Bruno
 #' @examples
-#' paragraph <- c("This is an example.", "Great code uses examples like this example.")
-#' paramatch(paragraph, 2)
-paramatch <- function(input, top_n_words = 5, suppress.input = FALSE) {
+#' paragraph <- data.frame(id = c(1:2), text = c("This is an example.", "Great code uses examples like this example."), stringsAsFactors = FALSE)
+#' paramatch(paragraph, "text", 2)
+
+#' @export
+paramatch <- function(dataframe, col, top_n_words = 5, suppress.input = FALSE, blacklist = c()) {
+  # Grab the string vector
+  paragraph_col <- dataframe[[col]]
   # Make sure inputs are valid
-    stopifnot(is.character(input))
-    stopifnot(is.numeric(top_n_words))
-    stopifnot(top_n_words >= 1)
-    stopifnot(top_n_words %% 1 == 0)
-    stopifnot(is.logical(suppress.input))
+  stopifnot(is.character(paragraph_col))
+  stopifnot(is.numeric(top_n_words))
+  stopifnot(top_n_words >= 1)
+  stopifnot(top_n_words %% 1 == 0)
+  stopifnot(is.logical(suppress.input))
   # Standardize the input
-    paragraph <- toupper(input)
-    paragraph <- gsub("[[:punct:]]", " ", paragraph)
-    paragraph <- gsub("[[:space:]]+", " ", paragraph)
-  # Split string into words  
-    allwords <- unlist(strsplit(paragraph, " "))
+  paragraph_col <- toupper(paragraph_col)
+  paragraph_col <- gsub("[[:punct:]]", " ", paragraph_col)
+  paragraph_col <- gsub("[[:space:]]+", " ", paragraph_col)
+  # Munge
+  if(!('top_n' %in% names(inputs))) { # Train # TODO add something in case top N length is differnt from inputs
+    # Find the top N words
+    # Split string into words  
+    allwords <- unlist(strsplit(paragraph_col, " "))
     allwords <- allwords[allwords != ""]
-  # Split into unique words
+    # Split into unique words
     words <- unique(allwords)
-  # Full word matching
+    # Full word matching
     frequency <- data.frame(word = words, 
                             occurances = sapply(words, function(x) length(grep(paste0("\\<",x,"\\>"), allwords))), 
                             stringsAsFactors = FALSE) 
-  # Ensure that N isn't greater than the total number of unique words
+    # Blacklist
+    if (length(blacklist) >0) {
+      for (i in 1:length(blacklist)) {
+        frequency <- frequency[!frequency$word == blacklist[[i]],]
+      }
+    }
```

Comment:
you can also replace the whole block with something like 
`frequency <- frequency[!frequency$word %in% blacklist]`